app/post.go

```diff
@@ -396,6 +397,11 @@ func (a *App) FillInPostProps(post *model.Post, channel *model.Channel) *model.A
 		post.DelProp("channel_mentions")
 	}
 
+	matched, err := regexp.MatchString(`\B@`, post.Message)
```

Comment:
Won't this detect any mention and most regular usage of the at sign? Would it be worth it to explicitly check for group mentions

Reply:
Yea it would result in this prop being set anytime there is a single `@` in the message body - do you think delaying post creation and querying `groups` by `name` for all matched mentions in the post body and then applying the prop only if it matches an existing ldap group?

Reply:
^^ That's what I was thinking. We could explicitly check for a group mention but it means more computation and possibly a delay in the call. I'm not sure if it's worth it ... But any thoughts on that Harrison?

Reply:
Farhan and I chatted about this, and I'm now in agreement that this is the better way to do it. On one hand, it's adding extra data to every post containing a mention that's made by a user that doesn't have this permission, but at the same time, that's likely a lot less work than we'd require to verify the message actually contains a group mention.

Reply:
How frequently is this going to be called ? If it's not rare, then it might be a good idea to pre-compile the regex and store it somewhere for better performance.

Reply:
@agnivade 
Thanks for the suggestion. I've implemented this suggestion! 😄